connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/InboundConnectorManager.java

```diff
@@ -156,7 +164,15 @@ private void activateConnector(InboundConnectorDefinitionImpl newConnector) {
             "Cannot activate webhook connector. "
                 + "Check whether property camunda.connector.webhook.enabled is set to true.");
       }
-      executable.activate(inboundContext);
+      if (executable instanceof PollingConnectorExecutable) {
+        Long processDefinitionKey = newConnector.processDefinitionKey();
+        Supplier<List<ProcessInstance>> processInstancesSupplier =
+            () -> processDefinitionSearch.fetchProcessInstancesWithVariables(processDefinitionKey);
+        executable.activate(
+            new InboundIntermediateConnectorContextImpl(inboundContext, processInstancesSupplier));
+      } else {
+        executable.activate(inboundContext);
+      }
```

Comment:
Suggestion to simplify

```suggestion
      if (executable instanceof PollingConnectorExecutable) {
        Long processDefinitionKey = newConnector.processDefinitionKey();
        Supplier<List<ProcessInstance>> processInstancesSupplier =
            () -> processDefinitionSearch.fetchProcessInstancesWithVariables(processDefinitionKey);
        inboundContext = new InboundIntermediateConnectorContextImpl(inboundContext, processInstancesSupplier);
      }
      executable.activate(inboundContext);
```